### DISCUSSION TITLE
ensuring calendar event show up event when javascript is disabled (sveltekit)

### DISCUSSION THREAD

**Discussion by max-208** [vkurko/calendar#162] [Q&A]
hi, i'm currently developing a SvelteKit app that still needs to be usable even when JavaScript is disabled, I'm having some issues as i can't get events to show up in a DayGrid when client side JavaScript is disabled.

minimal reproduction example : [https://github.com/max-208/svelte-calendar-not-working-js-disabled](https://github.com/max-208/svelte-calendar-not-working-js-disabled)
steps to reproduce : 
- download the repo, install dependancies and launch it
- go to the served url, javascript is enabled, the calendar loads the events
![image](https://github.com/vkurko/calendar/assets/29046242/1e56aa56-5d35-4cf9-93fe-a4b88fb5a54c)
- disable javascript in dev tools, the calendar no longer loads the events (this is not just visual, they are not present in the generated html)
![image](https://github.com/vkurko/calendar/assets/29046242/4d005506-5c88-4331-a32a-8d1e3c88a26f)

is there something that can be done so that the events are rendered server-side ?

**Comment by vkurko** [comment]
Apparently, in the current version, the calendar will not be able to fully render on the server side. To do this, I will have to rethink some of the solutions applied to optimize the rendering of events, as well as the operation of some settings that assume the presence of the DOM. In the future, I will think about what can be done about this.

**Comment by vkurko** [comment]
I'll just mark the tickets I found - [2259](https://github.com/sveltejs/svelte/issues/2259), [4848](https://github.com/sveltejs/svelte/issues/4848) - that may also affect the ssr issue on the calendar.
